### PR TITLE
feat: add row_count to chunks system table

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
I want to compute the storage consumption of data in IOx (bytes per row). While the row count exists in `system.columns` the DataFusion JOIN functionality is not good enough yet for me to calculate storage statistics directly (I can’t join in the table to get row counts).

Here is the join error for the morbidly curious:
```
select *
FROM
chunks JOIN columns using (database_name, partition)
limit 10;

Error running observer query: Error running observer query: Error during planning: Schema contains duplicate unqualified field name 'partition_key'
```

### CODE REVIEW

File: server/src/db/system_tables.rs [hunk 1]
```diff
@@ -111,38 +112,17 @@ fn from_chunk_summaries(chunks: Vec<ChunkSummary>) -> Result<RecordBatch> {
...
-            Arc::new(partition_key),
-            Arc::new(table_name),
-            Arc::new(storage),
-            Arc::new(estimated_bytes),
-            Arc::new(time_of_first_write),
-            Arc::new(time_of_last_write),
-            Arc::new(time_closing),
-        ],
-    )
+    RecordBatch::try_from_iter(vec![
```
Comment (alamb) [inline server/src/db/system_tables.rs:115]: This uses the new code I added in https://github.com/apache/arrow-rs/pull/7 to reduce the ceremony needed to create a `RecordBatch`

**Review by domodwyer — APPROVED** [review]
I enjoyed having a poke around the table summary stuff - we should definitely be able to automate some of this :)

**Comment by alamb** [comment]
>  we should definitely be able to automate some of this :)

@domodwyer  -- which parts to do you have in mind?

**Comment by domodwyer** [comment]
We can probably wire up a few proc macros to make (semi) arbitrary structs into `RecordBatches`

```rust
#[derive(SystemTable("chunks")]
struct ChunkSummaryTable(Vec<ChunkSummary>);

#[derive(SystemTableRow)]
struct ChunkSummary { .. }
```

It gets a bit more tricky if you need to do transformations on the data though, maybe field attributes could work around it - I think someone looked at using Serde for something similar in this area? What was the outcome?

**Comment by alamb** [comment]
> We can probably wire up a few proc macros to make (semi) arbitrary structs into RecordBatches

...

I had a hacky `Serde` implementation that could serialize to a system table for some types of data; I believe @tustvold  tried to take my crappy code and make it work and decided it was easier to do this. Maybe it is time to revisit that decision

**Pull request merged**